### PR TITLE
Simplify/update Activity BP

### PR DESCRIPTION
**Pull request by dweinstein**
looking for +1's before merge... maybe squash before merge too

### CODE REVIEW

File: android/2014-06-04-check-activities.md [hunk 1]
```diff
@@ -38,10 +38,10 @@ public static Intent getStartingIntent(Context context,
 
 Avoid intent filters on Activities if they are private, instead use explicit intent. 
 
-```
+```xml
 <activity
 	  android:name="com.app.YourActivity"
-    android:label="@string/app_name"
+	  android:label="@string/app_name"
```
Comment (radare) [inline android/2014-06-04-check-activities.md]: tab+spaces? should be just one tab like the other lines

Comment (dweinstein) [inline android/2014-06-04-check-activities.md]: fixed in bc28fb9

File: android/2014-06-04-check-activities.md [hunk 2]
```diff
@@ -8,15 +8,15 @@ categories:
 order: 603
 ---
 
-## Details 
+## Details
 
-Typically in Android applications an Activity is a “Screen” in an app. An activity can be invoked by any application if it is exported, thus allowing the user access to regions of the application often in unintended ways. If you define an Intent filter for an Activity it will automatically make it public and accessible by other applications. Even if an Activity is not public it can still be invoked with root privileges. This could allow an attacker to jump around an application in a way the developer did not intend, such as jumping past a password lock screen to access the data.
+An Activity can be invoked by any application if it is [`exported` and `enabled`](http://developer.android.com/guide/topics/manifest/activity-element.html). This could allow an attacker to load UI elements in a way the developer may not intend, such as jumping past a password lock screen to access data or functionality. By default Activities are not exported, however, if you define an Intent filter for an Activity, by default it will automatically be exported by the system.
```
Comment (poliva) [inline android/2014-06-04-check-activities.md]: there are two "by default" in the last sentence. I'd probably remove the second one.

Comment (dweinstein) [inline android/2014-06-04-check-activities.md]: fixed in c0a77dc

**Comment by giantpune** [comment]
+1

**Comment by Fuzion24** [comment]
how about "Activities can be directly invoked with unsanitized data" -> "Activities can be directly invoked and can include unsanitized data"

The first time I read it, it sounded like the unsanitized data was doing the activity invocation.

**Comment by dweinstein** [comment]
@Fuzion24 oh. I think of the call `startActivity(Intent)` where the Intent holds the potentially unsanitized data, therefore the "Activity can be invoked with ..." -- does that make sense?

**Comment by Fuzion24** [comment]
ah, yea.  I was thinking about on the receiving end of an activity: pulling extras out of the bundle.  This could also potentially unsanitized input

**Pull request merged**